cpp/include/raft/distance/detail/sparse_l2_nn.cuh

```diff
@@ -0,0 +1,303 @@
+/*
```

Comment:
Maybe rename file to `maked_l2_nn.cuh` (or even just `masked_nn.cuh` since it's new. I'd like to eventually deprecate `fused_l2_nn.cuh` in favor of `fused_nn.cuh`).

Reply:
Renamed to masked_nn.cuh (all other files as well)

Reply:
To make deprecating the `l2` versions easier, I have renamed the files to the `masked_nn*.cuh` format.